libcaf_test/caf/test/runnable.hpp

```diff
@@ -70,6 +70,91 @@ class CAF_TEST_EXPORT runnable {
                              stringify(rhs), location);
   }
 
+  /// Checks whether `lhs` and `rhs` are unequal.
+  template <class T0, class T1>
+  void check_ne(const T0& lhs, const T1& rhs,
+                const detail::source_location& location
+                = detail::source_location::current()) {
+    if (std::is_integral_v<T0> && std::is_integral_v<T1>) {
+      static_assert(std::is_signed_v<T0> == std::is_signed_v<T1>,
+                    "comparing signed and unsigned integers is unsafe");
+    }
```

Comment:
This is repeated in pretty much all of the checks. I think this should be factored out into a (private) function.